sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -511,17 +511,18 @@ object RewriteCorrelatedScalarSubquery extends Rule[LogicalPlan] with AliasHelpe
    * (first part of returned value), the HAVING clause of the innermost query block
    * (optional second part) and the parts below the HAVING CLAUSE (third part).
```

Comment:
Looks like this comment needs to be updated - in the new case it's returning None, rather than the inner query block below HAVING (and this is ok because we only needed the aggregate to fix the COUNT bug). Right?

Reply:
Yea, we should at least explain when the third part can be None.